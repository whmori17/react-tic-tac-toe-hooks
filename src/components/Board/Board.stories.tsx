import * as React from 'react';
import { Board } from '.';
import { Moves } from '@customTypes/Move';

export default {
  title: 'Board',
};

const onClick = (i: number) => console.log(i);
const emptySquares: Moves = ['', '', '', '', '', '', '', '', ''];
const squares: Moves = ['X', '', '', 'O', '', 'X', '', '', ''];

export const empty = () => <Board {...{ onClick, squares: emptySquares }} />;
export const withMoves = () => <Board {...{ onClick, squares }} />;
