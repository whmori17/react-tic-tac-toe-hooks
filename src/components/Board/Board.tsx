import React from 'react';
import { Square } from '..';
import { Moves } from '@customTypes/Move';

export interface BoardProps {
  squares: Moves;
  onClick: (position: number) => void;
}

export const Board: React.FC<BoardProps> = ({ squares, onClick }) => {
  const boardRows = [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8],
  ];

  return (
    <div className="game-board">
      {boardRows.map((row, index) => (
        <div className="board-row" key={index}>
          {row.map(position => (
            <Square value={squares[position]} onClick={() => onClick(position)} key={position} />
          ))}
        </div>
      ))}
    </div>
  );
};

Board.displayName = 'Board';
