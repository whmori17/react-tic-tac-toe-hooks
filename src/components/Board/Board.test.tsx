import * as React from 'react';
import { shallow } from 'enzyme';
import { Board, BoardProps } from './Board';

describe('Board', () => {
  const boardProps: BoardProps = {
    squares: ['', '', ''],
    onClick: (i: number) => console.log(i),
  };

  it('renders', () => {
    const wrapper = shallow(<Board {...boardProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
