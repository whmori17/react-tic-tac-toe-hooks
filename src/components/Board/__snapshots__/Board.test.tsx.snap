// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders 1`] = `
<div
  className="game-board"
>
  <div
    className="board-row"
    key="0"
  >
    <Square
      key="0"
      onClick={[Function]}
      value=""
    />
    <Square
      key="1"
      onClick={[Function]}
      value=""
    />
    <Square
      key="2"
      onClick={[Function]}
      value=""
    />
  </div>
  <div
    className="board-row"
    key="1"
  >
    <Square
      key="3"
      onClick={[Function]}
    />
    <Square
      key="4"
      onClick={[Function]}
    />
    <Square
      key="5"
      onClick={[Function]}
    />
  </div>
  <div
    className="board-row"
    key="2"
  >
    <Square
      key="6"
      onClick={[Function]}
    />
    <Square
      key="7"
      onClick={[Function]}
    />
    <Square
      key="8"
      onClick={[Function]}
    />
  </div>
</div>
`;
